### DISCUSSION TITLE
Coqui Software Libraries, Samples and API

### DISCUSSION THREAD

**Discussion by AdamSobieski** [coqui-ai/TTS#1670] [General Q&A] [answered]
Hello. Coqui STT provides `libkenlm.so` and `libstt.so` across platforms and includes some native client examples. I found them to be quite useful for [a project that I'm working on](https://github.com/AdamSobieski/Nifty) – in particular the .NET samples – and I wonder whether there is or will be some similar dynamically-linked libraries and native client samples for Coqui TTS?

I'm hoping to provide and to utilize some software API's for local installation, localhost, and remote service scenarios involving Coqui STT and TTS. I'm envisioning providing some interfaces with different implementations for those scenarios, local installation and service client.

In source code, something like:
```cs
ICoquiSTT stt;

if (settings.UseLocalInstallation)
{
    stt = STT.CreateLocalInstallation(settings);
}
else if (settings.UseClient)
{
    stt = STT.CreateClient(settings);
}
```
and similarly for TTS.

The Coqui STT dynamically-linked libraries, native client samples, and software API is simplifying the STT part of the indicated approach. I would like to ask about similar libraries, samples, and API for Coqui TTS. Thank you.

**Comment by erogol** [comment, marked as answer]
Thanks for the post. However we don't plan to provide such APIs any time soon.
